src/components/MainActionButton.tsx

```diff
@@ -7,7 +7,8 @@ export type Variant =
 	| 'secondary'
 	| 'positive'
 	| 'negative'
-	| 'alternate';
+	| 'alternate'
+	| 'disabled';
```

Comment:
We shouldn't make disabled a variant, but rather there should be a disabled prop on the MainActionButton. What you could do though, is in the logic of the MainActionButton, if the button is disabled, override the variant that the user provided. Otherwise use the variant the user provided (or default if neither is provided).

This allows the logic of what variant to use to be within the button itself.

Reply:
Or you can omit the variant altogether and use the CSS state:

https://www.w3schools.com/cssref/sel_disabled.asp